tests/test_auth_cli.py

```diff
@@ -0,0 +1,154 @@
+from unittest.mock import Mock
+
+import pytest
+
+from ggshield.cmd import cli
+from ggshield.config import Config
+
+from .conftest import my_vcr
+
+
+@pytest.fixture(autouse=True)
+def tmp_config(monkeypatch, tmp_path):
+    monkeypatch.setattr("ggshield.config.get_auth_config_dir", lambda: str(tmp_path))
+
+
+class TestAuthLoginToken:
+    @pytest.mark.parametrize(
+        "cassette, expect_success",
+        [
+            ("test_auth_login_token_invalid", False),
+            ("test_auth_login_token_invalid_scope", False),
+            ("test_auth_login_token_valid", True),
+        ],
+    )
+    def test_auth_login_token(self, cli_fs_runner, cassette, expect_success):
+        """
+        GIVEN an API token, valid or not
+        WHEN the auth login command is called with --method=token
+        THEN the validity of the token should be checked, and if valid, the user should be logged in
+        """
```

Comment:
I don't understand how this test works: did you edit the cassettes by hand to produce the expected responses? If so, that sounds dangerous: I would expect to be able to rm all cassettes to update them.

Reply:
Yes, but I don't see how I could do it like you suggest without using valid tokens. I could do what I did for the other tests however, which is mocking the entire API call without relying to cassettes.

Reply:
I think it would be better to mock, indeed. Maybe you can use a requests mocker, like [requests-mock](https://github.com/jamielennox/requests-mock)?

I am also starting to think maybe the details of the network calls should be in pygitguardian, what do you think? :thinking:

Reply:
I did a PR on pygitguardian ([here](https://github.com/GitGuardian/py-gitguardian/pull/30)) but the idea was rejected because the endpoint is not supposed to be public (it won't be documented in the public documentation).

I'll reuse the monkeypatch solution I already did in the other tests for now :slightly_smiling_face:

Reply:
Ah OK, I am not sure I agree with the reasoning, but monkeypatching sounds good for now :+1:.